test/lightning_web/live/credential_live_test.exs

```diff
@@ -761,6 +761,434 @@ defmodule LightningWeb.CredentialLiveTest do
+
+      assert edit_live |> has_element?("span", "Test User")
+    end
+
+    @tag :capture_log
+    test "failing to retrieve userinfo", %{
+      user: user,
+      bypass: bypass,
+      conn: conn
+    } do
+      expect_wellknown(bypass)
+
+      expect_userinfo(
+        bypass,
+        Lightning.AuthProviders.Common.get_wellknown!(:salesforce),
+        {400,
+         """
+         {
+           "error": "access_denied",
+           "error_description": "You're not from around these parts are ya?"
+         }
+         """}
+      )
+
+      expires_at = DateTime.to_unix(DateTime.utc_now()) + 3600
+
+      credential =
+        credential_fixture(
+          user_id: user.id,
+          schema: "salesforce_oauth",
+          body: %{
+            access_token: "ya29.a0AVvZ...",
+            refresh_token: "1//03vpp6Li...",
+            expires_at: expires_at,
+            scope: "scope1 scope2",
+            instance_url: "http://localhost:#{bypass.port}/auth/.well-known"
+          }
+        )
+
+      {:ok, edit_live, _html} = live(conn, ~p"/credentials")
+
+      assert wait_for_assigns(edit_live, :error, credential.id, "salesforce")
+
+      edit_live |> render()
+
+      assert edit_live
+             |> has_element?("p", "Failed retrieving your information.")
+
+      # Now respond with success
+      expect_userinfo(
+        bypass,
+        Lightning.AuthProviders.Common.get_wellknown!(:salesforce),
+        """
+        {"picture": "image.png", "name": "Test User"}
+        """
+      )
+
+      edit_live |> element("a", "try again.") |> render_click()
+
+      assert wait_for_assigns(edit_live, :userinfo, credential.id, "salesforce")
+
+      assert edit_live |> has_element?("span", "Test User")
+    end
+
+    @tag :capture_log
+    test "renewing an expired but invalid token", %{
+      user: user,
+      bypass: bypass,
+      conn: conn
+    } do
+      expect_wellknown(bypass)
+
+      expect_token(
+        bypass,
+        Lightning.AuthProviders.Common.get_wellknown!(:salesforce),
+        {400,
+         """
+         {
+           "error": "access_denied",
+           "error_description": "You're not from around these parts are ya?"
+         }
+         """}
+      )
+
+      expires_at = DateTime.to_unix(DateTime.utc_now()) - 50
+
+      credential =
+        credential_fixture(
+          user_id: user.id,
+          schema: "salesforce_oauth",
+          body: %{
+            access_token: "ya29.a0AVvZ...",
+            refresh_token: "1//03vpp6Li...",
+            expires_at: expires_at,
+            scope: "scope1 scope2",
+            instance_url: "http://localhost:#{bypass.port}/auth/.well-known"
+          }
+        )
+
+      {:ok, edit_live, _html} = live(conn, ~p"/credentials")
+
+      assert wait_for_assigns(edit_live, :error, credential.id, "salesforce")
+
+      edit_live |> render()
+
+      assert edit_live
+             |> has_element?("p", "Failed renewing your access token.")
```

Comment:
I'm curious how there is a test for failure but there is a missing code coverage line in `lib/lightning/auth_providers/common.ex:86`. Do you know why?